azurerm/internal/services/apimanagement/api_management_api_diagnostic_resource.go

```diff
@@ -158,3 +310,50 @@ func resourceArmApiManagementApiDiagnosticDelete(d *schema.ResourceData, meta in
 
 	return nil
 }
+
+func expandApiManagementApiDiagnosticHTTPMessageDiagnostic(input []interface{}) *apimanagement.HTTPMessageDiagnostic {
+	if len(input) == 0 {
+		return nil
+	}
+
+	v := input[0].(map[string]interface{})
+
+	result := &apimanagement.HTTPMessageDiagnostic{
+		Body: &apimanagement.BodyDiagnosticSettings{},
+	}
+
+	if bodyBytes, ok := v["body_bytes"]; ok {
+		result.Body.Bytes = utils.Int32(bodyBytes.(int32))
```

Comment:
Thanks @sirlatrom, one more suggestion after testing

```suggestion
		result.Body.Bytes = utils.Int32(int32(bodyBytes.(int)))
```

Reply:
Better now?